MetaMask/metamask-extension
# Fix #15050 - MV3: Keep the user logged in when service worker restarts

## Explanation

Currently, when the service worker restarts, the user gets logged out.  This PR is the extension portion of the login fix.  More details will be added as we get further along.

## More Information

* Fixes #15050
* See: #67890

## Depends on:

* `eth-keyring-controller` (https://github.com/MetaMask/KeyringController/pull/152)

## Manual Testing Steps

- Start the extension in mv3 mode: `yarn start:mv3`
- Login to MetaMask
- Lock MetaMask
- Log in again
- Open the DevTools "Application" panel and kill the service worker (it will restart)
- Execute actions that would require login *without refreshing the page* ("Create Account"), for example
- See that you're still logged in, actions working correctly

## Pre-Merge Checklist


### + If there are functional changes:
